src/ast/irbuilderbpf.cpp

```diff
@@ -326,6 +327,35 @@ CallInst *IRBuilderBPF::createMapLookup(const std::string &map_name,
   return createCall(lookup_func_type, lookup_func, { map_ptr, key }, name);
 }
 
+CallInst *IRBuilderBPF::createPerCpuMapLookup(const std::string &map_name,
+                                              Value *key,
+                                              Value *cpu,
+                                              const std::string &name)
+{
+  return createPerCpuMapLookup(map_name, key, cpu, GET_PTR_TY(), name);
+}
+
+CallInst *IRBuilderBPF::createPerCpuMapLookup(const std::string &map_name,
+                                              Value *key,
+                                              Value *cpu,
+                                              PointerType *val_ptr_ty,
+                                              const std::string &name)
+{
+  Value *map_ptr = GetMapVar(map_name);
+  // void *map_lookup_percpu_elem(struct bpf_map * map, void * key, u32 cpu)
+  // Return: Map value or NULL
+
+  assert(key->getType()->isPointerTy());
+  FunctionType *lookup_func_type = FunctionType::get(
+      val_ptr_ty, { map_ptr->getType(), key->getType(), getInt32Ty() }, false);
+  PointerType *lookup_func_ptr_type = PointerType::get(lookup_func_type, 0);
+  Constant *lookup_func = ConstantExpr::getCast(
+      Instruction::IntToPtr,
+      getInt64(libbpf::BPF_FUNC_map_lookup_percpu_elem),
+      lookup_func_ptr_type);
+  return createCall(lookup_func_type, lookup_func, { map_ptr, key, cpu }, name);
```

Comment:
I think we should add a `CreateHelperErrorCond` call here too

Reply:
We don't do this for `createMapLookup` because the return value is the map value or null not an error code.